ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -17,6 +17,121 @@
+         contents of the Genesis UTxO and finish.
+
+       - If there is a snapshot found, then deserialize the @state@ file, which
+         contains either:
+
+            - If the node was running __with the legacy database enabled__, it
+              contains a full LedgerState snapshot in the same style as before
+              the UTxO HD implementation (i.e. in-memory part of the LedgerState
+              (equal among implementations) + UTxO map).
+
+            - Otherwise, it contains a serialization of the in-memory part of the
+              LedgerState, without a UTxO map.
+
+        In both cases, the snapshot will be deserialized as @ExtLedgerState blk
+        EmptyMK@. In case we found an snapshot, we will overwrite the
+        @BackingStore@ tables with the @tables@ file from said snapshot.
+
+  3. Depending on whether the node is requested to run both databases and whether
+     or not the snapshot has a UTxO inside, the node will:
+
+       - __Run both__ + snapshot has UTxO inside: Extract the tables from the
+         deserialized snapshot to create a @ExtLedgerState blk ValuesMK@ which
+         can be 1. used in the legacy database and 2. after stripping out the
+         @ValuesMK@, used in the modern database.
+
+       - __Run both__ + snapshot doesn't have UTxO inside: an error will be
+         thrown as we cannot/don't think we should reconstruct an in-memory UTxO
+         directly from the @BackingStore@.
+
+       - Otherwise: Strip out the values inside the @ExtLedgerState blk EmptyMK@
+         to make sure that we don't have an in-memory UTxO map and use it as the
+         anchor for the modern database.
+
+  4. Replay on top of this ledger database all blocks up to the immutable
+     database tip.
+
+  At this point, the node carries a @LedgerDB@ that is initialized and ready to
+  be applied blocks on the volatile database.
+
+  == Taking snapshots during normal operation
+
+  Snapshots are taken by the @'copyAndSnapshotRunner'@ when the disk policy
+  dictates to do so. Whenever the chain grows past @k@ blocks, said runner will
+  copy the blocks which are more than @k@ blocks from the tip (i.e. the ones
+  that must be considered immutable) to the immutable database and then:
+
+  1. Flush differences in the modern DB up to the immutable db tip.
+
+  2. If dictated by the disk policy, serialize an @ExtLedgerState blk EmptyMK@ containing:
+
+       - If the legacy database is enabled, the state to serialize is the anchor
+         of the legacy database (which should be the same as the one from the
+         modern database) with the tables stowed.
+
+       - Otherwise, the state to serialize is the modern database anchor which
+         requires no further modifications.
+
+  3. A directory is created named after the slot number of the ledger state
+     being snapshotted, and the serialization from the previous step is written
```

Comment:
Isn't `3` part of `2` above?

Reply:
Not to me. 2 specifies what is going to be serialized, and 3 says where those things will be placed in the file system.